py/tests/unit/with_runtime_sparkling/test_pca.py

```diff
@@ -0,0 +1,58 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+from pyspark.ml import Pipeline, PipelineModel
+from pyspark.mllib.linalg import *
+from pyspark.mllib.linalg import *
+from pyspark.sql.types import *
+
+from tests import unit_test_utils
+from tests.unit.with_runtime_sparkling.algo_test_utils import *
+
+
+def testParamsPassedByConstructor():
+    assertParamsViaConstructor("H2OPCA")
+
+
+def testParamsPassedBySetters():
+    assertParamsViaSetters("H2OPCA")
+
+
+
+
+def getPreconfiguredAlgorithm():
+    return H2OPCA(k=4,
+                  maxIterations=500,
+                  pcaMethod="GramSVD",
+                  seed=42,
+                  transform="standardize")
+
+def testPipelineSerialization(arrestsDataset):
```

Comment:
Can you add some test making assertion on a real values. (For instance to check that predictions are not the same for all rows in the dataset)?

Reply:
Will do, thanks!